filebeat/processor/add_kubernetes_metadata/indexing.go

```diff
@@ -43,19 +43,36 @@ func newLogsPathMatcher(cfg common.Config) (add_kubernetes_metadata.Matcher, err
 		logPath = logPath + "/"
 	}
 
+	logp.Debug("kubernetes", "logs_path matcher log path: %s", logPath)
+
 	return &LogPathMatcher{LogsPath: logPath}, nil
 }
 
+// Docker container ID is a 64-character-long hexadecimal string
+const containerIdLen = 64
+
 func (f *LogPathMatcher) MetadataIndex(event common.MapStr) string {
 	if value, ok := event["source"]; ok {
 		source := value.(string)
-		logp.Debug("kubernetes", "Incoming source value: ", source)
+		logp.Debug("kubernetes", "Incoming source value: %s", source)
 		cid := ""
 		if strings.Contains(source, f.LogsPath) {
-			//Docker container is 64 chars in length
-			cid = source[len(f.LogsPath) : len(f.LogsPath)+64]
+			sourceLen := len(source)
+			logsPathLen := len(f.LogsPath)
+
+			if f.LogsPath == "/var/log/containers/" && strings.HasSuffix(source, ".log") && sourceLen >= containerIdLen + 4 {
+				// In case of the Kubernetes log path "/var/log/containers/",
+				// the container ID will be located right before the ".log" ending.
+				containerIdEnd := sourceLen - 4
+				cid = source[containerIdEnd - containerIdLen : containerIdEnd]
+			} else if sourceLen >= logsPathLen + containerIdLen {
+				// In any other case, we assume the container ID will follow right after the log path.
+				cid = source[logsPathLen : logsPathLen + containerIdLen]
+			}
+			logp.Debug("kubernetes", "Using container id: %s", cid)
+		} else {
+			logp.Debug("kubernetes", "Error extracting container id - source value does not contain log path.")
```

Comment:
You will need to refactor this logic after adding length checks, in case of wrong length the error won't be logged at the moment